libhb/vce_common.c

```diff
@@ -221,5 +316,178 @@ int hb_vce_av1_available()
     return -1;
 }
 
+int hb_check_amfdec_available()
+{
+    #if HB_PROJECT_FEATURE_AMFDEC
+        return 1;
+    #else
+        return 0;
+    #endif
+}
+
+int hb_vce_are_filters_supported(hb_list_t *filters)
+{
+    return 0;
+}
+int hb_vce_dec_is_enabled(hb_job_t *job)
+{
+    return 0;
+}
+int hb_vce_sanitize_filter_list(hb_job_t *job)
+{
+    return 0;
+}
 
 #endif // HB_PROJECT_FEATURE_VCE
+const char* hb_vce_decode_get_codec_name(enum AVCodecID codec_id)
+{
+    switch (codec_id)
+    {
+        case AV_CODEC_ID_H264:
+            return "h264_amf";
+
+        case AV_CODEC_ID_HEVC:
+            return "hevc_amf";
+
+        case AV_CODEC_ID_AV1:
+            return "av1_amf";
+
+        default:
+            return NULL;
+    }
+
+}
+int hb_vce_hw_filters_via_video_memory_are_enabled(hb_job_t *job)
+{
+    return job->hw_pix_fmt == AV_PIX_FMT_AMF_SURFACE;
+}
+hb_buffer_t *  hb_vce_copy_avframe_to_video_buffer(hb_job_t *job, AVFrame *frame, AVRational time_base)
```

Comment:
Is this function needed? Looks the same as hb_avframe_to_video_buffer

Reply:
The hw frame to sw frame conversion happens before, so there shouldn't be any need to check for sw_filters or whatever.